### PR TITLE
fix(Topology): check the number of block CMVs wrt the number of compo…

### PR DESCRIPTION
**Pull request by panquez**
…nents

### CODE REVIEW

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 1]
```diff
@@ -51,9 +54,109 @@ namespace geode
...
-                    unique_vertex_index, Block3D::component_type_static() ) );
+            const auto block_cmvs = brep_.component_mesh_vertices(
+                unique_vertex_index, Block3D::component_type_static() );
+            const auto block_uuids = components_uuids( block_cmvs );
+            if( block_uuids.size() == 1 )
+            {
+                const auto line_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Line3D::component_type_static() );
+                const auto surface_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Surface3D::component_type_static() );
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:65]: They are also used if size is 2, you can initialize them before the if statement

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 2]
```diff
@@ -51,9 +54,109 @@ namespace geode
...
+                            if( verbose_ )
+                            {
+                                Logger::info( "Unique vertex with index ",
+                                    unique_vertex_index,
+                                    " is part of one block (",
+                                    block_uuids.front().string(),
+                                    "), and an internal surface (",
+                                    surface_cmv.component_id.id().string(),
+                                    ") but no line, with several component "
+                                    "mesh vertices along the surface." );
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:87]: This shouldn't be a problem, the problem is if there is a line boundary only to this surface and several cmvs on the same surface

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 3]
```diff
@@ -51,9 +54,109 @@ namespace geode
...
+                        }
+                        return false;
+                    }
+                    return true;
+                }
+                if( brep_.has_component_mesh_vertices( unique_vertex_index,
+                        Corner3D::component_type_static() ) )
+                {
+                    // Skip since it requires to implement a local radial sort
+                    // to count number of block components
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:117]: Not sure about that.

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 4]
```diff
@@ -51,9 +54,109 @@ namespace geode
...
+                    return true;
+                }
+                absl::flat_hash_set< uuid > incident_surfaces_of_free_lines;
+                bool inside_free_line{ false };
+                for( const auto& line_cmv : line_cmvs )
+                {
+                    const auto& line = brep_.line( line_cmv.component_id.id() );
+                    if( brep_.nb_incidences( line_cmv.component_id.id() ) == 1
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:125]: line_cmv.component_id.id() -> line.id()

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 5]
```diff
@@ -51,9 +54,109 @@ namespace geode
...
+                    return true;
+                }
+                if( block_cmvs.size() != 1 )
+                {
+                    if( verbose_ )
+                    {
+                        Logger::info( "Unique vertex with index ",
+                            unique_vertex_index, " is part of one block (",
+                            block_uuids.front().string(),
+                            "), one internal surfaces, and on a free line, "
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:150]: You didn't check if the vertex was part of only one internal surface.

**Comment by BotellaA** [comment]
You need to update your branch with the latest master/next

**Review by BotellaA — APPROVED** [review]

**Comment by panquez** [comment]
> You need to update your branch with the latest master/next

Done!

File: src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp [hunk 6]
```diff
@@ -99,6 +116,70 @@ namespace geode
...
+                if( nb_internal_surface_cmvs == 0 )
+                {
+                    if( nb_block_cmvs != 1 )
+                    {
+                        if( verbose_ )
+                        {
+                            Logger::info( "Unique vertex with index ",
+                                unique_vertex_index, " is part of the block ",
+                                block_uuid.string(),
+                                " and no internal surface with ", nb_block_cmvs,
```
Comment (MelchiorSchuh) [inline src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp:144]:
```suggestion
                                " and none of its internal surfaces but has ", nb_block_cmvs,
```

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**